### ISSUE TITLE
【Question】The loss of the model drops very slowly

### ISSUE THREAD

**Issue by 5idaidai** [OpenNMT/OpenNMT-py#978]
I train a Text Summarization model, now run 135050 steps, and then the loss drops very slowly

```
[2018-10-09 10:17:18,118 INFO] Validation accuracy: 32.9939
[2018-10-09 10:45:17,061 INFO] Validation perplexity: 69.6471
[2018-10-09 10:45:17,070 INFO] Validation accuracy: 32.8886
[2018-10-09 11:15:09,439 INFO] Validation perplexity: 66.4453
[2018-10-09 11:15:09,448 INFO] Validation accuracy: 33.6084
[2018-10-09 11:42:51,936 INFO] Validation perplexity: 66.6906
[2018-10-09 11:42:51,945 INFO] Validation accuracy: 33.3187
[2018-10-09 12:12:46,440 INFO] Validation perplexity: 65.6247
[2018-10-09 12:12:46,450 INFO] Validation accuracy: 33.7751
[2018-10-09 12:40:44,725 INFO] Validation perplexity: 66.4654
[2018-10-09 12:40:44,734 INFO] Validation accuracy: 33.4591
[2018-10-09 13:08:50,452 INFO] Validation perplexity: 66.3592
[2018-10-09 13:08:50,462 INFO] Validation accuracy: 33.5732
[2018-10-09 21:09:50,220 INFO] Validation perplexity: 61.0705
[2018-10-09 21:09:50,224 INFO] Validation accuracy: 34.3544
[2018-10-09 22:16:38,878 INFO] Validation perplexity: 61.6657
[2018-10-09 22:16:38,887 INFO] Validation accuracy: 34.3632
[2018-10-09 23:23:20,031 INFO] Validation perplexity: 61.2705
[2018-10-09 23:23:20,041 INFO] Validation accuracy: 34.5124
[2018-10-10 00:30:46,289 INFO] Validation perplexity: 61.1886
[2018-10-10 00:30:46,299 INFO] Validation accuracy: 34.2579
[2018-10-10 01:47:09,623 INFO] Validation perplexity: 60.8182
[2018-10-10 01:47:09,632 INFO] Validation accuracy: 34.4949
[2018-10-10 02:55:31,891 INFO] Validation perplexity: 61.1988
[2018-10-10 02:55:31,901 INFO] Validation accuracy: 34.3895
[2018-10-10 04:04:57,864 INFO] Validation perplexity: 60.6381
[2018-10-10 04:04:57,874 INFO] Validation accuracy: 34.5739
[2018-10-10 05:13:04,052 INFO] Validation perplexity: 59.837
[2018-10-10 05:13:04,061 INFO] Validation accuracy: 34.6529
[2018-10-10 06:21:19,581 INFO] Validation perplexity: 60.1015
[2018-10-10 06:21:19,591 INFO] Validation accuracy: 34.4685
[2018-10-10 07:29:32,655 INFO] Validation perplexity: 60.4005
[2018-10-10 07:29:32,664 INFO] Validation accuracy: 34.5826
[2018-10-10 08:38:22,751 INFO] Validation perplexity: 60.1597
[2018-10-10 08:38:22,761 INFO] Validation accuracy: 34.5651
[2018-10-10 09:48:01,116 INFO] Validation perplexity: 60.2344
[2018-10-10 09:48:01,126 INFO] Validation accuracy: 34.3983
[2018-10-10 10:58:11,404 INFO] Validation perplexity: 60.1113
[2018-10-10 10:58:11,413 INFO] Validation accuracy: 34.3808
``` 

is this normal ?

my model's parameter is 

```
python -u train.py -data *** \
                   -save_model *** \
                   -train_from *** \
                   -copy_attn \
                   -global_attention mlp \
                   -word_vec_size 128 \
                   -rnn_size 512 \
                   -layers 1 \
                   -encoder_type brnn \
                   -train_steps 1000000 \
                   -max_grad_norm 2 \
                   -dropout 0. \
                   -batch_size 32 \
                   -optim adagrad \
                   -learning_rate 0.15 \
                   -adagrad_accumulator_init 0.1 \
                   -reuse_copy_attn \
                   -copy_loss_by_seqlength \
                   -bridge \
                   -seed 777 \
                   -gpuid 0 \
                   -valid_steps 5000 \
                   -save_checkpoint_steps 10000 \
                   -keep_checkpoint 10 \
                   -report_every 10 \
                   -share_embeddings
```

and the valid data result is

```
{
  "rouge-l": {
    "p": 0.21409543234543274,
    "r": 0.25845827747935224,
    "f": 0.21338204016751455
  },
  "rouge-2": {
    "p": 0.09896625874125897,
    "r": 0.12533619931283188,
    "f": 0.10640826006995237
  },
  "rouge-1": {
    "p": 0.23424811022311062,
    "r": 0.28222007709184793,
    "f": 0.24745095821588708
  }
}
```

the dataset has 1200000 pairs with article and title

**Comment by vince62s** [comment]
is there a specific reason why you are using adagrad and also not the transformer ?
you will get better results with the transformer, look at the FAQ.
